documentation/src/docs/beginner_dev_guide.md

```diff
@@ -0,0 +1,200 @@
+
+When instantiating the EventStreamWriter above, we passed in a [UTF8StringSerializer](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/impl/UTF8StringSerializer.java) instance. Pravega uses a [Serializer](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/stream/Serializer.html) interface in its writers and readers to simplify the act of writing and reading an object’s bytes to and from streams. The [JavaSerializer](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/impl/JavaSerializer.java) can handle any `Serializable` object.
+
+# 4.3 Create a Pravega Event Reader and read the event back from the stream.
+
+Readers are associated with reader groups, which track the readers’ progress and allow more than one reader to coordinate over which segments they’ll read.
+A [ReaderGroupManager](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/admin/ReaderGroupManager.html) is used to create a new reader group on the Pravega Stream.
+
+
+The below snippet create a [ReaderGroup](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/stream/ReaderGroup.html)
+```java
+try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope("examples", controllerURI)) {
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of("examples", "helloStream"))
+                .disableAutomaticCheckpoints()
+                .build();
+        readerGroupManager.createReaderGroup("readerGroup", readerGroupConfig);
+}
+```
+We can attach an Pravega Event Reader to this reader group and read the data from the Pravega Stream `helloStream`. The below snippet creates an EventReader called `reader` and reads the value from the Pravega Stream.
+
+```java
+try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope("examples",
+                ClientConfig.builder().controllerURI(controllerURI).build());
+     EventStreamReader<String> reader = clientFactory.createReader("reader",
+                "readerGroup",
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build())) {
+        String event = reader.readNextEvent(5000).getEvent();
+        System.out.println(event);
+}
+```
+
+# 5. What’s next?
+This guide covered the creation of a application that writes and reads from Pravega. However, there is much more. We recommend continuing the journey by going through [Pravega-client-101](https://blog.pravega.io/2020/09/22/pravega-client-api-101/) and other samples present in the [Pravega Samples repo](https://github.com/pravega/pravega-samples).
```

Comment:
I see that you also want to include a more advanced dev guide document. Maybe we need to refer to it in this section.